csharp/ql/src/semmle/code/csharp/ir/implementation/unaliased_ssa/internal/SSAConstruction.qll

```diff
@@ -65,6 +65,30 @@ private module Cached {
     instruction instanceof ChiInstruction // Chis always have modeled results
   }
 
+  cached
+  predicate hasConflatedMemoryResult(Instruction instruction) {
+    instruction instanceof UnmodeledDefinitionInstruction
+    or
+    instruction instanceof AliasedDefinitionInstruction
+    or
+    instruction.getOpcode() instanceof Opcode::InitializeNonLocal
+    or
+    exists(OldInstruction oldInstruction | instruction = Chi(oldInstruction) |
+      Alias::getResultMemoryLocation(oldInstruction).getVirtualVariable() instanceof
+        Alias::AliasedVirtualVariable
+      or
+      // If there is no memory location for a memory result, then it's unmodeled
+      // and therefore conflated with every other unmodeled instruction.
+      oldInstruction.hasMemoryResult() and
+      not exists(Alias::getResultMemoryLocation(oldInstruction))
+    )
+    or
+    exists(Alias::MemoryLocation location |
+      instruction = Phi(_, location) and
+      location.getVirtualVariable() instanceof Alias::AliasedVirtualVariable
```

Comment:
I should be possible to have a `Phi` for a specific (non-conflated) location that happens to be part of `AliasedVirtualVariable`. I'm not sure if we have this specific test case, but in `ssa.cpp`, function `MergeMustExactlyWithMustTotallyOverlap` shows how this works when the location in question does not escape. You get two `Phi` instructions, one for the entire virtual variable, and one for the specific location within that virtual variable. See the QLDoc for the `PhiInsertion` module in `SSAConstruction.qll` for a bit more detail on the how and why.

Reply:
What would a test case look like?

Reply:
Perhaps this issue has to do with the test failures.

Reply:
I've written a test case that exhibits the problem you describe.

Reply:
I've just pushed a commit to demonstrate the problem, followed by a commit to fix it.